src/components/SingleArticle.jsx

```diff
@@ -1,33 +1,61 @@
-import { fetchArticleByID } from '../api-utils'
-import { useEffect, useState } from "react"
-import {useParams} from "react-router-dom"
+import { fetchArticleByID } from "../api-utils";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { patchVotes } from "../api-utils";
 
 const SingleArticle = () => {
-  const [currentArticle, setCurrentArticle] = useState({})
+  const [currentArticle, setCurrentArticle] = useState({});
+  const [isError, setIsError] = useState(false);
   const dateObj = new Date(currentArticle.created_at);
   const dateString = dateObj.toString();
-  const { article_id } = useParams()
-
+  const { article_id } = useParams();
+  const [userViewVotes, setUserViewVotes] = useState(0);
+  
 
   useEffect(() => {
     fetchArticleByID(article_id).then((article) => {
-      setCurrentArticle(article)
-    })
-  }, [article_id])
+      setCurrentArticle(article);
+    });
+  }, [article_id]);
 
+  const handleVoteClick = (increment) => {
+    setUserViewVotes(userViewVotes + increment);
+    patchVotes(currentArticle.article_id, increment).catch((err) => {
+      console.log(err);
+      setUserViewVotes(userViewVotes - increment);
+      setIsError(true);
+    });
+  };
 
-  return (
-    <div>
-    <h2>{currentArticle.title}</h2>
-    <h3> Author: {currentArticle.author} </h3>
-    <p>Created at: {dateString}</p>
-    <p>Topic: {currentArticle.topic}</p>
-      <p>{currentArticle.body}</p>
-    <p> Votes: {currentArticle.votes} </p>
-    <p> Comments: {currentArticle.comment_count}</p>
-    </div>
-  )
 
-}
+  if (isError) {
```

Comment:
If everything else is going to stay the same apart from the error message it's better to just conditionally render that message rather than everything. e.g.

```js
{isError ? <p>Sorry...</p> : null}
```